ml - add premium leads params to property model

magneland opened the pull request:
Bug fix: do not inject extra attributes during validation.
Rename validation method.
Add patch method to client.
Add an amount validator.
Add a company object.
'companyCapAmount' is assumed to be set to MAX_INT on demand in RentLinx.
Handcrafted changes to vcr cassette for testing.
Requires a corresponding change to the RentLinx API.
Only run Travis CI for 'ruby-2.2.3’.

TODO: change ’PUT’ to 'PATCH' once Rentlinx supports it
TODO: make tests run in RubyMine
TODO: fix tests to make vcr cassettes disposable

magneland committed:
ml - add premium leads params to property model

Bug fix: do not inject extra attributes during validation.
Rename validation method.
Add patch method to client.
Add an amount validator.
Add a company object.
'companyCapAmount' is assumed to be set to MAX_INT on demand in RentLinx.
Handcrafted changes to vcr cassette for testing.
Requires a corresponding change to the RentLinx API.
Only run Travis CI for 'ruby-2.2.3’.

TODO: change ’PUT’ to 'PATCH' once Rentlinx supports it
TODO: make tests run in RubyMine
TODO: fix tests to make vcr cassettes disposable

XanderStrike commented on .travis.yml:
```diff
@@ -1,16 +1,9 @@
-  - 2.0.0
-  - 2.1.5
-  - 2.1.6
-  - 2.2.2
```
We originally wanted to make sure it worked across a variety of ruby versions because this is an open source gem and others may want to use it.

Was something special done that is definitely not supported in older ruby versions?

magneland replied:
My comment got killed with the browser dying !!
Anyways, the answer is: no, so  I can add it back in

XanderStrike commented on lib/rentlinx/validators/attribute_processor_service.rb:
```diff
@@ -37,7 +38,9 @@ def process
-                     address: AddressValidator }
+                     address: AddressValidator,
+                     capAmount: AmountValidator,
+                     companyCapAmount: AmountValidator }
```
Should we validate rent and deposit as well?

magneland replied:
Sure

magneland replied:
Actually...that change should be somewhere else I think. It has some repercussions.

XanderStrike commented on test/cassettes/test_property-test_property_patch_method.yml:
```diff
@@ -0,0 +1,135 @@
+      - '54'
+    body:
+      encoding: UTF-8
+      string: '{"AccessToken":"F49B4B03-BBB1-4DF3-923A-038C730A7E03"}'
```
:frowning:

magneland replied:
Looks like it is in many files. Will investigate solutions.

magneland committed:
ml - don’t store access token in cassettes

The previous access token is obsolete anyways.
Also don’t use a bunch of different names for the same thing.

TODO: fix tests that currently depend on old/customized cassettes

magneland committed:
ml - revert travis config to previous ruby versions, but bump to 2.2.3

magneland committed:
ml - fix rubocop 0.35.1 offenses

XanderStrike commented:
:+1:
